lib/wasi/src/runners/wasi_common.rs

```diff
@@ -132,6 +136,31 @@ fn prepare_filesystem(
     Ok(Box::new(OverlayFileSystem::new(root_fs, [container_fs])))
 }
 
+/// HACK: We need this so users can mount host directories at relative paths.
+/// This assumes that the current directory when a runner starts will be "/", so
+/// instead of mounting to a relative path, we just mount to "/$path".
+///
+/// This isn't really a long-term solution because there is no guarantee what
+/// the current directory will be. The WASI spec also doesn't require the
+/// current directory to be part of the "main" filesystem at all, we really
```

Comment:
> *should* be mounting to a relative directory

Not quite sure what you mean there. 
Mounting to a relative directory doesn't make sense, because as you mentioned, the relative directory is an ephemeral thing that always changes in the instance environment.

As I see it, mounting only makes sense with both absolute host and guest paths. 
Auto-expanding a relative to an absolute path makes sense for convenience, but I don't see what else we should be doing.